sw/applications/example_im2col/im2col_lib.h

```diff
@@ -14,26 +15,66 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
-#include "im2colGolden.h"
+#include "im2col_golden.h"
+#include "im2col_input.h"
+#include "im2col.h"
 #include "dma.h"
+#include "im2col_spc_regs.h"
 #include "core_v_mini_mcu.h"
 #include "x-heep.h"
 #include "rv_plic.h"
 #include "csr.h"
+#include <math.h>
 
-// By default, printfs are activated for FPGA and for simulation.
+#include "mmio.h"
+#include "handler.h"
+#include "hart.h"
+#include "fast_intr_ctrl.h"
+#include "timer_sdk.h"
+
+/* 
+    Choose between several HW configurations:
+    - 0: Only CPU
+    - 1: Exploit standard DMA
+    - 2: Exploit 2D DMA
+    - 3: Exploit Smart Peripheral Controller (SPC)
+*/ 
+#define HW_CONFIG_CPU
+#define HW_CONFIG_DMA_1D
+#define HW_CONFIG_DMA_2D
+#define HW_CONFIG_SPC
+
+/* Defines which DMA channels are available to the SPC, depending on HW specifications */
+#define SPC_CH_MASK 0b0001 
+
+/* Defines the datatype of the input */
+#define INPUT_DATATYPE 0
+
+/* Base address of the im2col SPC */
+#define IM2COL_SPC_BASE_ADDR EXT_PERIPHERAL_START_ADDRESS + 0x4000
```

Comment:
@TommiTerza please, use the define in sw/device/lib/drivers/im2col_spc/im2col.h:57 instead of redefining it, so that it is less error prone.